### ISSUE TITLE
improve signature memory efficiency

### ISSUE THREAD

**Issue by drahnr** [Richterrettich/rpm-rs#11]
For large rpms, currently the content is copied.
For rpms with sizes beyond 100mb (the ones with many assets i.e. games) might suffer OOM conditions when running on tighter memory bounds.

Proposal: Migrate to a `trait Read` based approach, using a custom cursor which allows sequential read across multiple buffers (header + content).

**Comment by drahnr** [comment]
Yep, I should find a few hours to impl that plus do a repo structure refactor :)

**Comment by Richterrettich** [comment]
Nice, but please do the restructuring in a separate PR since it is easier to review then.
